modal/_utils/function_utils.py

```diff
@@ -190,6 +190,33 @@ def serialized_function(self) -> bytes:
             logger.debug(f"Serializing function for class service function {self.cls.__qualname__} as empty")
             return b""
 
+    def get_cls_vars(self) -> Dict[str, Any]:
+        if self.cls is not None:
+            cls_vars = {
+                attr: getattr(self.cls, attr)
+                for attr in dir(self.cls)
+                if not callable(getattr(self.cls, attr)) and not attr.startswith("__")
+            }
+            return cls_vars
+        return {}
+
+    def get_cls_var_attrs(self) -> Dict[str, Any]:
+        import dis
+
+        import opcode
+
+        LOAD_ATTR = opcode.opmap["LOAD_ATTR"]
+        STORE_ATTR = opcode.opmap["STORE_ATTR"]
+        ATTR_OPS = (LOAD_ATTR, STORE_ATTR)
+
+        func = self.raw_f
+        code = func.__code__
+        f_attr_ops = {instr.argval for instr in dis.get_instructions(code) if instr.opcode in ATTR_OPS}
```

Comment:
Just double-checking for my own understanding that `instr.argval` here will be the attribute *name* not attribute *value*.

Reply:
Yep. It's going to be the name of the attribute accessed in the opcode